pennylane_cirq/simulator_device.py

```diff
@@ -132,9 +132,10 @@ def apply(self, operations, **kwargs):
         # pylint: disable=missing-function-docstring
         super().apply(operations, **kwargs)
 
-        # We apply an identity gate to all wires, otherwise Cirq would ignore
+        # We apply identity gates to all wires, otherwise Cirq would ignore
         # wires that are not acted upon
-        self.circuit.append(cirq.IdentityGate(len(self.qubits))(*self.qubits))
```

Comment:
This line broke the PasqalDevice, since it has restrictions about how close qubits must be in order to have multi-qubit gates. Note that it would break any other cirq device in the future which also had topology restrictions

Reply:
I'm wondering if applying identity gates to unused qubits is the right approach. It sounds like Cirq is being somewhat smart here, and simply just discarding qubits from the simulation that are not involved in the computation. Pyquil does this same; we should be able to simply take into account these ignored qubits when returning the samples/probabilities.

Reply:
Yes, but I think this requires a change to how the devices work, doesn't it?

Reply:
I don't think so, the Forest simulators also act like this, and we take it into account in the PL plugin. We simply keep track of the inactive wires, and 'expand_dim' the probability before returning it.

Reply:
Yes, then it makes sense to do that here. I think the previous solution (identity on all wires) was a quick hack.

But it might also make sense to provide helper functions for keeping track of such things in the base Device class?